src/core/unit.c

```diff
@@ -209,12 +209,17 @@ int unit_add_name(Unit *u, const char *text) {
 
         if (unit_name_is_valid(text, UNIT_NAME_TEMPLATE)) {
 
-                if (!u->instance)
-                        return -EINVAL;
+                if (!u->instance) {
+                        r = -EINVAL;
+                        log_error_errno(r, "instance is NULL for name '%s': %m", text);
```

Comment:
This shouldn't log at error level, because it's a low-level "library" function. For example, 
unit_add_name() is called from unit_new_for_name() which doesn't log at all, which means that the caller logs.
So most likely you can only add logging at debug level here. Or make sure that the caller does not log, but that'd require other changes, to ensure that all code paths log once.

This should use use unit_log_*_errno.
For the user, a message like "instance is not set" is more appropriate. The fact that we internally store that as NULL is not relevant.
Also, SYNTHETIC_ERRNO.
So probably something like this:
```C
if (!u->instance)
     return log_unit_debug_errno(u, SYNTHETIC_ERRNO(EINVAL),
                                                   "instance is not set when adding name '%s'.", text);
```
(I'm don't remember how log_unit_* format the message, so please check that it looks OK.)

Reply:
Thank you very much. 
We have revised it according to your suggestion, please take a look again.